client/lib/titan/new-mailbox.js

```diff
@@ -0,0 +1,236 @@
+	} );
+
+const getMailboxRequiredStringValue = () =>
+	PropTypes.shape( {
+		value: PropTypes.string.isRequired,
+		error: PropTypes.string,
+	} ).isRequired;
+
+const getMailboxPropTypeShape = () =>
+	PropTypes.shape( {
+		uuid: PropTypes.string.isRequired,
+		alternativeEmail: getMailboxOptionalStringValue(),
+		domain: getMailboxRequiredStringValue(),
+		isAdmin: getMailboxRequiredBooleanValue(),
+		mailbox: getMailboxRequiredStringValue(),
+		name: getMailboxRequiredStringValue(),
+		password: getMailboxRequiredStringValue(),
+	} );
+
+const sanitizeEmailSuggestion = ( emailSuggestion ) =>
+	String( emailSuggestion )
+		.replace( /[^0-9a-z_'.-]/gi, '' )
+		.toLowerCase();
+
+const validateMailboxesAreUnique = ( mailboxes ) => {
+	const mailboxNameCounts = mailboxes.reduce( ( nameCount, mailbox ) => {
+		nameCount[ mailbox.mailbox ] = 1 + nameCount[ mailbox.mailbox ] ?? 0;
+		return nameCount;
+	}, {} );
+
+	return mailboxes.map( ( mailbox ) => {
+		if ( mailboxNameCounts[ mailbox.mailbox ] > 1 ) {
+			return {
+				...mailbox,
+				[ mailbox ]: { value: mailbox.mailbox, error: translate( 'Please use unique mailboxes' ) },
+			};
+		}
+		return mailbox;
+	} );
+};
+
+const mapMailboxFields = ( mailbox, callback, skipFields = [] ) => {
+	return Object.entries( mailbox )
+		.map( ( entry ) => {
+			const fieldName = entry[ 0 ];
+			if ( fieldName === 'uuid' || skipFields.includes( fieldName ) ) {
+				return entry;
+			}
+			const fieldValue = entry[ 1 ];
+			return [ fieldName, callback( fieldValue, fieldName ) ];
+		} )
+		.reduce( ( updatedMailbox, entry ) => {
+			updatedMailbox[ entry[ 0 ] ] = entry[ 1 ];
+			return updatedMailbox;
+		}, {} );
+};
+
+const removePreviousError = ( { value } ) => ( { value, error: null } );
+
+const clearPreviousErrors = ( mailboxes ) => {
+	return mailboxes.map( ( mailbox ) => mapMailboxFields( mailbox, removePreviousError ) );
+};
+
+const validateRequiredField = ( { value, error } ) => ( {
+	value,
+	error:
+		! error && typeof value !== 'boolean' && ( ! value || '' === value.trim() )
+			? translate( 'This field is required.' )
+			: error,
+} );
+
+const validateRequiredMailboxFields = ( mailbox, optionalFields = [] ) => {
+	return mapMailboxFields( mailbox, validateRequiredField, optionalFields );
+};
+
+const validateFullEmailAddress = ( { value, error }, allowEmpty = false ) => {
+	if ( error ) {
+		return { value, error };
+	}
+	if ( allowEmpty && value === '' ) {
+		return { value, error };
+	}
+	if ( emailValidator.validate( value ) ) {
+		return { value, error };
+	}
+	return {
+		value,
+		error: translate( 'Please supply a valid email address.' ),
+	};
+};
+
+const validateMailboxName = ( { value, error }, { value: domainName, error: domainError } ) => {
+	if ( error ) {
+		return { value, error };
+	}
+	if ( ! /^[0-9a-z_-](\.?[0-9a-z_-])*$/i.test( value ) ) {
+		return {
+			value,
+			error: translate( 'Only numbers, letters, dashes, underscores, and periods are allowed.' ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Only numbers, letters, dashes, underscores, apostrophes and periods are allowed.' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/titan-mailbox-provisioning-new-components)